src/components/Button.astro

```diff
@@ -1,21 +1,35 @@
 ---
-const { text, link } = Astro.props;
+import type { LinkProps } from "../types";
+
+interface Props extends LinkProps {
+    marginClass?: string;
+}
+
+const { text, link, marginClass } = Astro.props;
 ---
 
-<a class="button rounded-2xl" href={link}>{text}</a>
+<a class=`button rounded-2xl ${marginClass}` href={link}>{text}</a>
```

Comment:
This seems like how Tailwind might expect us to pass down classes that the parent component wants to add but open to better suggestions on this! https://tailwindcss.com/docs/content-configuration#dynamic-class-names

Reply:
Agreed, using Tailwind's dynamic classes like this should be the correct approach! I'd suggest changing it to 
`const { text, link, id, class: customClass } = Astro.props;` 
so we can use the regular `class` prop for any Tailwind classes, not just `marginClass`. This makes our component more flexible.

Reply:
Good call! Added in https://github.com/With-the-Ranks/withtheranks-astro/pull/15/commits/361ca1ac4165dcbd15e934cadaeb2c484efcb304